cvat-ui/src/containers/annotation-page/top-bar/top-bar.tsx

```diff
@@ -279,6 +281,49 @@ class AnnotationTopBarContainer extends React.PureComponent<Props> {
         this.unblock();
     }
 
+    private async handlePlayIfNecessary(): Promise<void> {
+        const {
+            jobInstance,
+            frameNumber,
+            frameDelay,
+            frameFetching,
+            playing,
+            canvasIsReady,
+            onSwitchPlay,
+            onChangeFrame,
+        } = this.props;
+
+        const { stopFrame } = jobInstance;
+
+        if (playing && canvasIsReady && !frameFetching && !this.isWaitingForPlayDelay) {
+            this.isWaitingForPlayDelay = true;
+            try {
+                await new Promise((resolve) => {
+                    setTimeout(resolve, frameDelay);
+                });
+
+                const { playing: currentPlaying } = this.props;
+
+                if (currentPlaying) {
+                    const nextCandidate = frameNumber + 1;
+                    if (nextCandidate > stopFrame) {
+                        onSwitchPlay(false);
+                        return;
+                    }
+
+                    const next = await jobInstance.frames.search({ notDeleted: true }, nextCandidate, stopFrame);
```

Comment:
It seems if `notDeleted` is `true`, deleted images will not be displayed, even when the "Show deleted frames" setting is enabled.

Reply:
good point